src/v3/src/components/OpenOktaVerifyFPButton/OpenOktaVerifyFPButton.tsx

```diff
@@ -61,14 +64,25 @@ const OpenOktaVerifyFPButton: UISchemaElementComponent<{
       variant: 'primary',
       wide: true,
       step,
-      onClick: () => setKey(key + 1),
+      onClick: () => {
+        const isAppLinkMethod = challengeMethod === CHALLENGE_METHOD.APP_LINK;
+        const isUniversalLinkMethod = challengeMethod === CHALLENGE_METHOD.UNIVERSAL_LINK;
+        if ((isAppLinkMethod || isUniversalLinkMethod) && href) {
+          if (isAndroid() && !isAppLinkMethod) {
+            Util.redirectWithFormGet(href);
+          } else {
+            window.location.assign(href);
+          }
+        }
+        setKey(key + 1);
+      },
     },
   };
 
   return (
     <React.Fragment>
       <Button uischema={buttonUiSchema} />
-      {href && (
+      {(href && challengeMethod === CHALLENGE_METHOD.CUSTOM_URI) && (
       <IFrame
```

Comment:
@shawyu-okta  Just to clarify is the IFrame only shown for the CUSTOM_URI method?